src/solvers/prop/prop.h

```diff
@@ -116,6 +116,20 @@ class propt:public messaget, public prop_assignmentt
     warning() << "CPU limit ignored (not implemented)" << eom;
   }
 
+  /// Set random variable seed
+  /// (supported by solvers such as minisat2 and glucose)
+  virtual void set_random_seed(std::size_t value)
+  {
+    warning() << "random-seed not supported" << eom;
+  }
+
+  /// Set random variable frequency
+  /// (supported by solvers such as minisat2 and glucose)
+  virtual void set_random_var_freq(double value)
+  {
+    warning() << "random-var-freq not supported" << eom;
+  }
```

Comment:
Design idea/question: instead of introducing a new `set_...` function for each future option, can't we have a `prop_configt` made up of `optionalt<T>` fields (presently an `optionalt<std::size_t>` and an `optionalt<double>`) and one `set_config(prop_configt)`? The default implementation here would then check each field for being `none`, and else yield a warning. That would also address one of my comments made elsewhere.

Reply:
I think @peterschrammel might want to answer this question.

Reply:
I am agaist introducing more and more custom configuration classes: They are likely to become incompatible with each other, requiring to copy/adapt configuration data at interfaces. 
I'm in favour of having a single dictionary-like data structure such as optionst, which is passed along throughout the codebase and carries all the configuration data. Optionst should be enhanced with proper 'optional' behaviour and a way to override individual items without requiring a full copy. @LAJW had a go at such an optionst replacement some time ago (https://github.com/diffblue/cbmc/pull/1319).

For this PR, I could propose to get rid of set_* functions and replace them by an optionst-based solution.

Reply:
Isn't `satcheck_infot` such a custom configuration class? I was just thinking to move this to `propt`. My rationale being the present need to add additional `set_...` calls at each site whenever new ones are added, plus the problem that there may be spurious warnings as said  in my other comment: https://github.com/diffblue/cbmc/pull/1501#discussion_r159768942

Reply:
@peterschrammel Map from #1319 is an overkill. If you want dictionaries, just use `std::unordered_map`. I'm in favour of structs, as they lean towards parsing done in one spot, rather than spreading throughout the entire application (they separate UI from business logic). It's bad enough that we print to standard output all over the place, I didn't want to contribute to the problem.

`satcheck_infot` was a common satcheck constructor argument in one of the earlier revisions of this PR. If we truly cannot change satcheck constructor signature, then putting that as a `set_config` method (which would serve the role of a constructor) might be the next best thing.

Reply:
> It's bad enough that we print to standard output all over the place, I didn't want to contribute to the problem.

Pointers?

Reply:
The fact that large amount of classes inherits from `messaget`.

Reply:
So the `message_handlert` decides how to log -- I'm not sure why that's a problem? Pick a different message handler if you need to log differently?

Reply:
It doesn't matter what is the name of the class to which we're printing. The fact that we're printing raw strings right next to the business logic is the problem.

Reply:
So what would you be doing instead?

Reply:
If it were up to me, I'd build a view instead and delegate formatting to that. View would be passed as argument or functions could return what to print to view as their result. Views could easily be swapped (CLI/JSON/Dummy, etc.). But it's not up to me.

Reply:
Constructive ideas will always be appreciated, rambling less so. Everyone should feel empowered to create an issue to trigger discussion, or directly start with a pull request (at the risk of it being handed back for major rework).

Supposedly it is up to a message handler to take care of formatting. Surely that isn't fully done this way and needs fixing (hence, e.g., #1359). Yet your comments sound like you see a need for much bigger changes?

Reply:
> So what *would* you be doing instead?
You've requested my opinion, I gave it to you. Yes, slightly bigger changes than just hotfixes.

But I think we've strayed much too far away from the original point, and it's up to @peterschrammel to decide how he would like argument passing to look like. Current state is what he specifically requested, so I'm not going to go against his design.